pkg/client/client_test.go

```diff
@@ -374,6 +375,51 @@ var _ = Describe("ConmonClient", func() {
 				Expect(result).NotTo(BeNil())
 				Expect(result.TimedOut).To(Equal(true))
 			})
+
+			testName = "should handle many requests"
+			if terminal {
+				testName += " with terminal"
+			}
+			FIt(testName, func() {
+				createRuntimeConfigWithProcessArgs(terminal, []string{"/busybox", "sleep", "30"})
+
+				sut = configGivenEnv(tmpDir, rr.runtimeRoot, terminal)
+				resp, err := sut.CreateContainer(context.Background(), &client.CreateContainerConfig{
+					ID:         ctrID,
+					BundlePath: tmpDir,
+					Terminal:   terminal,
+				})
+				Expect(err).To(BeNil())
+				Expect(resp.PID).NotTo(Equal(0))
+				Eventually(func() error {
+					return rr.RunCommandCheckOutput(ctrID, "list")
+				}, time.Second*5).Should(BeNil())
+
+				// Start the container
+				Expect(rr.RunCommand("start", ctrID)).To(BeNil())
+
+				// Wait for container to be running
+				Eventually(func() error {
+					return rr.RunCommandCheckOutput("running", "list")
+				}, time.Second*10).Should(BeNil())
+
+				var wg sync.WaitGroup
+				for i := 0; i < 15; i++ {
+					go func() {
+						wg.Add(1)
```

Comment:
the wg.Add should be prior to the go func since it'll race.

Reply:
ah good to know